source/extensions/filters/http/grpc_json_transcoder/json_transcoder_filter.h

```diff
@@ -126,6 +127,7 @@ class JsonTranscoderConfig : public Logger::Loggable<Logger::Id::config>,
 
   Protobuf::DescriptorPool descriptor_pool_;
   google::grpc::transcoding::PathMatcherPtr<MethodInfoSharedPtr> path_matcher_;
+  mutable absl::Mutex type_helper_mutex_;
   std::unique_ptr<google::grpc::transcoding::TypeHelper> type_helper_;
```

Comment:
Can you add a locked by annotation on this object to make sure we hit all uses now and in the future?

Reply:
It is too complicate to protect all type_helper_.   its resolver is already thread-safe.   Only its type_info_ is not. 

I changed code here to make type_info thread safe.

https://github.com/grpc-ecosystem/grpc-httpjson-transcoding/pull/64